### PR TITLE
Update contributing docs with pointers for legacy checkpoint testing

### PR DESCRIPTION
**Pull request by ninginthecloud**
## What does this PR do?

This PR updates the testing session in CONTRIBUTING.md by adding scripts to download saved checkpoints from AWS storage. Without these checkpoints, multiple local tests will fail. 

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [X] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -195,6 +195,16 @@ python -m pip install ".[dev, examples]"
 python -m pip install pre-commit
 ```
 
+Additionally, for testing back compatibility with legacy versions of Pytorch Lightning, you also need to download all saved version-checkpoints from the public AWS storage. Run the following script to get all saved version-checkpoints:
```
Comment (ananthsub) [inline .github/CONTRIBUTING.md:198]:
```suggestion
Additionally, for testing backward compatibility with older versions of Pytorch Lightning, you also need to download all saved version-checkpoints from the public AWS storage. Run the following script to get all saved version-checkpoints:
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -195,6 +195,16 @@ python -m pip install ".[dev, examples]"
...
+
+```bash
+cd legacy
+wget https://pl-public-data.s3.amazonaws.com/legacy/checkpoints.zip
+unzip -o checkpoints.zip
+```
+
+Note: These checkpoints are generated to set baselines for maintaining back-compatibility with legacy versions of Pytorch Lightning. Details of checkpoints for back-compatibility can be found [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/legacy/README.md).
```
Comment (ananthsub) [inline .github/CONTRIBUTING.md:206]:
```suggestion
Note: These checkpoints are generated to set baselines for maintaining backward compatibility with legacy versions of Pytorch Lightning. Details of checkpoints for back-compatibility can be found [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/legacy/README.md).
```

**Review by ananthsub — APPROVED** [review]
thanks for adding this @ninginthecloud !

**Review by kaushikb11 — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -195,6 +195,16 @@ python -m pip install ".[dev, examples]"
 python -m pip install pre-commit
 ```
 
+Additionally, for testing backward compatibility with older versions of Pytorch Lightning, you also need to download all saved version-checkpoints from the public AWS storage. Run the following script to get all saved version-checkpoints:
+
+```bash
+cd legacy
+wget https://pl-public-data.s3.amazonaws.com/legacy/checkpoints.zip
```
Comment (ananthsub) [inline .github/CONTRIBUTING.md:202]: might want to clarify which directory these must be downloaded to

**Review by tchaton — APPROVED** [review]
LGMT !

**Review by Borda — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -195,6 +195,16 @@ python -m pip install ".[dev, examples]"
 python -m pip install pre-commit
 ```
 
+Additionally, for testing backward compatibility with older versions of Pytorch Lightning, you also need to download all saved version-checkpoints from the public AWS storage. Run the following script to get all saved version-checkpoints:
```
Comment (awaelchli) [inline .github/CONTRIBUTING.md:198]:
```suggestion
Additionally, for testing backward compatibility with older versions of PyTorch Lightning, you also need to download all saved version-checkpoints from the public AWS storage. Run the following script to get all saved version-checkpoints:
```

Comment (awaelchli) [reply .github/CONTRIBUTING.md:198]: I would like to also suggest to formulate it as optional. I believe to get started with contributing this is *absolutely not necessary*.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -195,6 +195,16 @@ python -m pip install ".[dev, examples]"
...
+
+```bash
+cd legacy
+wget https://pl-public-data.s3.amazonaws.com/legacy/checkpoints.zip
+unzip -o checkpoints.zip
+```
+
+Note: These checkpoints are generated to set baselines for maintaining backward compatibility with legacy versions of Pytorch Lightning. Details of checkpoints for back-compatibility can be found [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/legacy/README.md).
```
Comment (awaelchli) [inline .github/CONTRIBUTING.md:206]:
```suggestion
Note: These checkpoints are generated to set baselines for maintaining backward compatibility with legacy versions of PyTorch Lightning. Details of checkpoints for back-compatibility can be found [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/legacy/README.md).
```

**Review by awaelchli — APPROVED** [review]
great!

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (ananthsub) [reply .github/CONTRIBUTING.md:198]: @awaelchli - in that case, this should be reflected with `make test` too: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md#developments-scripts 

or `make test` should not run legacy checkpoint tests

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (ninginthecloud) [reply .github/CONTRIBUTING.md:202]: yep. I can update this like 
```bash
wget https://pl-public-data.s3.amazonaws.com/legacy/checkpoints.zip -P legacy/
unzip -o  legacy/checkpoints.zip -d legacy/
```

**Pull request merged**